security/security_tools.md

```diff
@@ -0,0 +1,179 @@
+
+### [CodeQL](https://codeql.github.com/)
+
+CodeQL is a hosted service provided by GitHub which provides vulnerability scanning on behalf of GitHub users. While GitHub provides their dedicated Advanced Security service separately from the CodeQL project, the documentation within the CodeQL project confirms that CodeQL libraries provide the underlying architecture for this Advanced Security solution.
+While CodeQL provides vulnerability scanning features which are automated for GitHub repositories, there currently exist no implementation of any integration of CodeQL with any alternative version control projects dependent upon git (such as GitLab).
+CodeQL itself contains various extensions to the underlying rulesets which is offered for a set of default languages and frameworks. More specifically, the following languages and frameworks are supported: C, C++, C#, Go, Java, Kotlin, JavaScript, Python, Ruby, Swift, and TypeScript. While there exist GitHub repositories through which one may be able to submit support for additional languages, the size of the user community (which includes GitHub itself) will ensure that the requirements for integrating CodeQL are fairly high. Beyond this, CodeQL is typically integrated into any given GitHub repository by modifying specific repository settings (this is described in the [community documentation](https://docs.github.com/en/code-security/code-scanning/introduction-to-code-scanning/about-code-scanning)). As such, end-users will then receive the vulnerability scanning reports when commits have been pushed to the repository, or in response to a pull request having been opened.
+
+### [Semgrep](https://semgrep.dev/)
+
+Semgrep has an open source version and a paid version (which includes additional rules and a software composition analyzer).  We evaluated the open source version.
+
+To enable Semgrep, you can set up a CircleCi job using the Semgrep docker image.
+
+-   [Example PR in allsearch-frontend](https://github.com/pulibrary/allsearch_frontend/commit/a25b0f2b7c4294b68ff57a3c23fd721da91d1223)
+
+-   [Example PR with custom rules in bibdata](https://github.com/pulibrary/bibdata/pull/2189/files)
+
+To run it locally, you can install it with brew.
+
+Semgrep supports [many languages](https://semgrep.dev/docs/supported-languages/), including Ruby, Javascript, Typescript, PHP, and Python.  There are [43 rules for Ruby](https://semgrep.dev/p/ruby) and [71 rules for Javascript](https://semgrep.dev/p/javascript).  It can run on linux or macOs.  It can also run on Windows via Windows Subsystem for Linux or through the official docker image.
+
+Semgrep is very fast, and makes it easy to create custom rules.  Its VSCode integration works well.
+
+Exploitation Enumeration Solutions
+----------------------------------
+
+These allow a human user to send requests to a running application to identify potential vulnerabilities.
+
+### [Burp Suite](https://portswigger.net/burp/documentation/desktop)
```

Comment:
Recommendation: add Dastardly to this heading since that's how it's referenced in the table above